OMCompiler/Compiler/MidCode/MidCodeUtil.mo

```diff
@@ -0,0 +1,365 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-2018, Open Source Modelica Consortium (OSMC),
+ * c/o Linköpings universitet, Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF GPL VERSION 3 LICENSE OR
+ * THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ * ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+ * RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ * ACCORDING TO RECIPIENTS CHOICE.
+ *
+ * The OpenModelica software and the Open Source Modelica
+ * Consortium (OSMC) Public License (OSMC-PL) are obtained
+ * from OSMC, either from the above address,
+ * from the URLs: http://www.ida.liu.se/pr OpenModelica distribution.
+ * GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
+ *
+ * This program is distributed WITHOUT ANY WARRANTY; without
+ * even the implied warranty of  MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+ * IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+ *
+ * See the full OSMC Public License conditions for more details.
+ *
+*/
+
+encapsulated package MidCodeUtil
+protected
+import MidCode;
+import SimCode.{RecordDeclaration};
+import DAE;
+import DAE.{AvlTreePathFunction};
+import DAEDump;
+import FCore;
+import HashTableStringToPath;
+import Absyn;
+import List;
+public
+type cacheValue = AvlTreePathFunction.Value;
+
+/*Functions for experimental MidCode cache. Not Included for now.*/
```

Comment:
I know it is a bad practice to leave commented out code. My plan here is to get it in. Then make a separate PL to remove the comments but keep it in the history. I plan to discard my old repository

Reply:
I also think, that if I would include the JIT I should make the cache in a different way.
What I did just to try it out was forcing it into the DAE function tree (Works but results in a confusing design since the cache all of a sudden got dual responsibilities) (Maybe a C++ side cache would be the best option later... The current plan is just to get things in)